src/plasmapy/simulation/particle_integrators.py

```diff
@@ -150,3 +153,71 @@ def boris_push(x, v, B, E, q, m, dt, inplace: bool = True):
     else:
         v = vplus + hqmdt * E
         return x + v * dt, v
+
+
+def boris_push_relativistic(x, v, B, E, q, m, dt, inplace: bool = True):
+    r"""
+    The explicit Boris pusher, including relativistic corrections.
+
+    Parameters
+    ----------
+    x : `~np.ndarray`
+        particle position at full timestep, in SI (meter) units.
+    v : `~np.ndarray`
+        particle velocity at half timestep, in SI (meter/second) units.
+    B : `~np.ndarray`
+        magnetic field at full timestep, in SI (tesla) units.
+    E : float
+        electric field at full timestep, in SI (V/m) units.
+    q : float
+        particle charge, in SI (Coulomb) units.
+    m : float
+        particle mass, in SI (kg) units.
+    dt : float
+        timestep, in SI (second) units.
+
+    Notes
+    -----
+    For the basic overview of this algorithm, see `boris_push`. This
+    version, based on [1]_, applies relativistic corrections such as
+    TODO.
```

Comment:
TODO

Reply:
I added some more details here. Let me know what you think! (I have next to no experience with relativity)

Reply:
Looks good to me!